Artem/client

borodulinartm opened the pull request:
Код ревью перед 2 дедлайном

borodulinartm committed:
backup

borodulinartm committed:
Фича: добавление протокола

borodulinartm committed:
Рефакторинг

borodulinartm committed:
В протокол добавлены получатели

borodulinartm committed:
Доработка клиента

borodulinartm committed:
Резервная копия

AleksMVP commented on QTcp_client/dialog.cpp:
```diff
@@ -28,105 +32,121 @@ void Dialog::onSokDisplayError(QAbstractSocket::SocketError socketError) {
-    if (is_button_clicked) {
-        is_button_clicked = false;
-        QByteArray buffer;
+    if (isButtonClicked) {
```
По поводу этого в телеге написал.

AleksMVP commented on QTcp_client/dialog.cpp:
```diff
@@ -28,105 +32,121 @@ void Dialog::onSokDisplayError(QAbstractSocket::SocketError socketError) {
+        QString buffer;
 
-        QDataStream sock_stream(_sok);
+        QDataStream sock_stream(socket);
```
Кстати идея добавлять поля класса _ - была хорошей.

AleksMVP commented on QTcp_client/dialog.cpp:
```diff
@@ -28,105 +32,121 @@ void Dialog::onSokDisplayError(QAbstractSocket::SocketError socketError) {
-            QString message = QString("%1 : No messages in server").arg(_sok->socketDescriptor());
-            emit newMessage(message);
-            return;
+        for(;;) {
```
Лично для меня while(true) смотрится в разы красивее. Это чисто мои стилевые соображения, можешь не обращать на них внимания)

AleksMVP commented on QTcp_client/dialog.cpp:
```diff
@@ -28,105 +32,121 @@ void Dialog::onSokDisplayError(QAbstractSocket::SocketError socketError) {
+                        addToLog(message.getReceiverUser() + ": " + message.getMessage(), Qt::blue);
+                    }
+                } else {
+                    displayError("Incorrect Data!!!");
```
Возможно в этом месте стоит поставить как-то дебаг лог т.к. если нам сервер начинает слать битые данные это что-то странное.

borodulinartm committed:
Оптимизация под базу данных + изменения в протоколе

borodulinartm committed:
Протокол переписан с использованием JSON

borodulinartm committed:
Реализация отправки файлов на базовом уровне

borodulinartm committed:
Добавлена поддержка отправки файлов

borodulinartm committed:
Мелкие изменения

borodulinartm committed:
Поддержка базы данных

borodulinartm committed:
Подготовил клиента для интеграции с авторизацией

borodulinartm committed:
Небольшая чистка репозитория

borodulinartm committed:
НОВОЕ: статус сообщений (требует тестирования)

borodulinartm committed:
НОВОЕ: добавлена информация, в сети ли пользователь

borodulinartm committed:
Чистка репозитория

borodulinartm committed:
НОВОЕ: если юзер отключился, а потом снова подключился, то сообщения у него отобразятся
